compiler/rustc_codegen_ssa/src/mir/intrinsic.rs

```diff
@@ -331,6 +344,56 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                 use crate::common::AtomicOrdering::*;
                 use crate::common::{AtomicRmwBinOp, SynchronizationScope};
 
+                let invalid_monomorphization = |ty| {
+                    bx.tcx().dcx().emit_err(InvalidMonomorphization::BasicIntegerType {
+                        span,
+                        name,
+                        ty,
+                    });
+                };
+
+                let parse_const_generic_ordering = |ord: ty::Value<'tcx>| {
+                    let discr = ord.valtree.unwrap_branch()[0].unwrap_leaf();
+                    let ord = discr.to_atomic_ordering();
+                    // We have to translate from the intrinsic ordering to the backend ordering.
+                    use std::intrinsics::AtomicOrdering;
```

Comment:
You should probably just embed the discriminants directly in cg_ssa. The libstd cg_ssa is compiled against may use different discriminants from the one that rustc will end up compiling.

Reply:
I followed what we already do for `cmp::Ordering` where we also rely on the discriminants to be the same.

Reply:
`Ordering` is stable with guaranteed discriminant values, `AtomicOrdering` is not stable and I assume never will be, or at least not in the current location.

Reply:
`AtomicOrdering` got specifically added for this very purpose. It's going to be exactly as stable as we need it for this purpose.

Reply:
I still don't like this. I did rather just duplicate the discriminant values into the compiler and add a comment that both definitions need to be kept in sync.

Reply:
I think that's strictly worse than what I did. Duplicating the values has a much bigger chance of them diverging than if they are all in the same place.